### PR TITLE
Add more information to the error code.

### PR DESCRIPTION
**Pull request by csukuangfj**

### CODE REVIEW

File: modules/core/include/opencv2/core/base.hpp [hunk 1]
```diff
@@ -79,18 +79,18 @@ enum Code {
     BadImageSize=              -10,  //!< image size is invalid
     BadOffset=                 -11,  //!< offset is invalid
     BadDataPtr=                -12,  //!<
-    BadStep=                   -13,  //!<
+    BadStep=                   -13,  //!< image step is wrong
     BadModelOrChSeq=           -14,  //!<
-    BadNumChannels=            -15,  //!<
+    BadNumChannels=            -15,  //!< bad number of channels
```
Comment (alalek) [inline modules/core/include/opencv2/core/base.hpp]: Probably it is "not wrong", neither "invalid" or "bad" - current  value is usually "not supported"/"unsupported" by current implementation.

Comment (csukuangfj) [inline modules/core/include/opencv2/core/base.hpp]: I copied the information from https://github.com/opencv/opencv/blob/master/modules/core/src/system.cpp#L764

In this case, does the function `cvErrorStr` also need to be updated, returning "supported/unsupported" ?

**Comment by vpisarev** [comment]
@csukuangfj, since you've started it, can you please also add comments to all other error codes?

**Comment by csukuangfj** [comment]
@vpisarev 
Thank you for the review.

I have added comments to every enums that are used in the source code.

Some of the enums are defined but never used and I do not understand its exact meaning, so they are left
uncommented.

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**